### ISSUE TITLE
keywords as property names

### ISSUE THREAD

**Issue by bhelx** [dylibso/xtp-rust-bindgen#17]
Sometimes when using a keyword as a property name, it explodes. We could mangle the names, but would it make sense to just prepend every property with `r#` ?

```
#[derive(Serialize, Deserialize, Debug)]
struct Foo {    
    r#a_nullable: Option<i32>,
    r#type: String,
}
```

**Comment by zshipko** [comment]
Good idea - we should probably use `r#` for function names too.

**Comment by chrisdickinson** [comment]
It's probably worth keeping a table of which names _need_ to be mangled since using `r#` identifiers is a little less ergonomic. (I'm sure there's prior art out there in the proc-macro ecosystem, I'll do a little looking)

**Comment by zshipko** [comment]
It looks like it's possible to use normal identifiers even when the field is declared using a raw identifier: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5ea04ad4ee94b4789dacab33e4ea4afe

But autocomplete may suggest the raw name instead which may be a little confusing.

**Comment by zshipko** [comment]
Closed by #18
